### PR TITLE
fix : change the docker-registry secret's name to be configurable.

### PR DESCRIPTION
**Pull request by skarltjr**
## Description 
in this PR (#3321), createDockerRegistrySecret() makes a secret named "harbor-secret"
I think the name of secret should be "docker-registry-secret"

++ 
now createDockerRegistrySecret() makes a secret named "container-image-registry-secret"
also we can create docker-registry-secret with secretName param

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java [hunk 1]
```diff
@@ -444,7 +444,7 @@ public static Secret createDockerRegistrySecret(String dockerServer, String user
     String dockerConfigAsStr = Serialization.jsonMapper().writeValueAsString(dockerConfigMap);
 
     return new SecretBuilder()
-        .withNewMetadata().withName("harbor-secret").endMetadata()
+        .withNewMetadata().withName("docker-registry-secret").endMetadata()
```
Comment (rohanKanojia) [inline kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java:447]: Let's make the secret name configurable. Could you please add another method parameter `secretName`

**Review by rohanKanojia — APPROVED** [review]

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java (hunk 1 above)
Comment (skarltjr) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java]: @rohanKanojia  thx😄
could you merge this?

**Review by sunix — CHANGES_REQUESTED** [review]

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java [hunk 2]
```diff
@@ -423,13 +423,14 @@ private static Class<?> loadRelated(Class<?> type, String suffix, Class<?> defau
...
-   * 
+   *
    * @param dockerServer User to store key value pair for auths map
    * @param username username that needs to be used during secret creation
    * @param password password that needs to be used during secret creation
+   * @param secretName secretName that needs to be used during secret creation
    * @return an object of Secret
    */
-  public static Secret createDockerRegistrySecret(String dockerServer, String username, String password)
+  public static Secret createDockerRegistrySecret(String dockerServer, String username, String password, String secretName)
```
[the remark this answers is not in the document]
Comment (skarltjr) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java:454]: @sunix 
could you check this again?

**Comment by skarltjr** [comment]
@sunix
could you check this again?🙂

now we can create docker-registry-secret with default name( "docker-registry-secret") or secretName param

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java [hunk 3]
```diff
@@ -423,14 +424,46 @@ private static Class<?> loadRelated(Class<?> type, String suffix, Class<?> defau
...
    * @param username username that needs to be used during secret creation
    * @param password password that needs to be used during secret creation
    * @return an object of Secret
    */
   public static Secret createDockerRegistrySecret(String dockerServer, String username, String password)
       throws JsonProcessingException {
+    Map<String, Object> dockerConfigMap = createDockerRegistryConfigMap(dockerServer, username, password);
+    String dockerConfigAsStr = Serialization.jsonMapper().writeValueAsString(dockerConfigMap);
+
+    return new SecretBuilder()
```
Comment (sunix) [inline kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java:438]: OK but now you have code duplication, here I would simply call 
```suggestion
    return createDockerRegistrySecret(dockerServer, username, password, DEFAULT_DOCKER_REGISTRY_SECRET_NAME);
```

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java [hunk 4]
```diff
@@ -423,14 +424,46 @@ private static Class<?> loadRelated(Class<?> type, String suffix, Class<?> defau
...
+        .withNewMetadata().withName(DEFAULT_DOCKER_REGISTRY_SECRET_NAME).endMetadata()
+        .withType("kubernetes.io/dockerconfigjson")
+        .addToData(".dockerconfigjson", Base64.getEncoder().encodeToString(dockerConfigAsStr.getBytes(StandardCharsets.UTF_8)))
+        .build();
```
Comment (sunix) [inline kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java:442]: OK but now you have code duplication, here I would simply call 
```suggestion
    return createDockerRegistrySecret(dockerServer, username, password, DEFAULT_DOCKER_REGISTRY_SECRET_NAME);
```

Comment (sunix) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java:442]: also, where is `createDockerRegistrySecret` called?

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java [hunk 5]
```diff
@@ -440,13 +473,6 @@ public static Secret createDockerRegistrySecret(String dockerServer, String user
...
     dockerConfigMap.put("auths", auths);
-
-    String dockerConfigAsStr = Serialization.jsonMapper().writeValueAsString(dockerConfigMap);
-
-    return new SecretBuilder()
-        .withNewMetadata().withName("harbor-secret").endMetadata()
-        .withType("kubernetes.io/dockerconfigjson")
-        .addToData(".dockerconfigjson", Base64.getEncoder().encodeToString(dockerConfigAsStr.getBytes(StandardCharsets.UTF_8)))
-        .build();
+    return dockerConfigMap;
```
Comment (sunix) [inline kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java:476]: I am not sure to understand how this is replaced with a new code with a default name (or custom)

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java [hunk 6]
```diff
@@ -50,6 +50,7 @@ private KubernetesResourceUtil() {
 
   public static final Pattern KUBERNETES_DNS1123_LABEL_REGEX = Pattern.compile("[a-z0-9]([-a-z0-9]{0,61}[a-z0-9])?");
   private static final Pattern INVALID_LABEL_CHARS_PATTERN = Pattern.compile("[^-A-Za-z0-9]+");
+  private static final String DEFAULT_DOCKER_REGISTRY_SECRET_NAME = "docker-registry-secret";
```
Comment (sunix) [inline kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java:53]: Minor: I wouldn't use the word `docker` in the name as it is just one possible registry. What about `container-image-registry-secret` ?

Comment (skarltjr) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java]: Thank you for your good suggestion!

I removed the duplication and use more general name as you suggest!

Comment (skarltjr) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java]: @sunix
Thank you for your good suggestion!

I removed the duplication and used more general name as you suggest

**Comment by skarltjr** [comment]
@manusa 
I forgot to check code style. sorry 😅
could you check this again?

**Comment by skarltjr** [comment]
thx 😄
@manusa @rohanKanojia @sunix

**Pull request merged**